app/consumer.py

```diff
@@ -13,21 +25,34 @@ def __init__(self):
         super(Consumer, self).__init__()
 
     def on_message(self, unused_channel, basic_deliver, properties, body):
-        logger.info("Received message", queue=self.QUEUE, delivery_tag=basic_deliver.delivery_tag, app_id=properties.app_id)
 
-        try:
-            mongo_id = body.decode("utf-8")
-            document = get_doc_from_store(mongo_id)
+        delivery_count = get_delivery_count_from_properties(properties)
 
-            processor = CoraProcessor(logger, document, self._ftp)
+        logger.info(
+            'Received message',
+            queue=self.QUEUE,
+            delivery_tag=basic_deliver.delivery_tag,
+            delivery_count=delivery_count,
+            app_id=properties.app_id
+        )
+
+        mongo_id = body.decode("utf-8")
+        document = get_doc_from_store(mongo_id)
+        processor = CoraProcessor(logger, document, self._ftp)
+
+        try:
+            processor.process()
+            self.acknowledge_message(basic_deliver.delivery_tag, tx_id=processor.tx_id)
+            logger.info("Processed successfully", tx_id=processor.tx_id)
 
-            processed_ok = processor.process()
-            if processed_ok:
-                logger.info("Processed successfully", tx_id=processor.tx_id)
-                self.acknowledge_message(basic_deliver.delivery_tag, tx_id=document['tx_id'])
+        except BadMessageError as e:
+            # If it's a bad message then we have to reject it
+            self.reject_message(basic_deliver.delivery_tag, tx_id=processor.tx_id)
+            logger.error("Bad message", action="rejected", exception=e, delivery_count=delivery_count, tx_id=processor.tx_id)
 
-        except Exception as e:
-            logger.error("ResponseProcessor failed", exception=e, tx_id=document['tx_id'])
+        except (RetryableError, Exception) as e:
+            self.nack_message(basic_deliver.delivery_tag, tx_id=processor.tx_id)
+            logger.error("Failed to process", action="nack for retry", exception=e, delivery_count=delivery_count, tx_id=processor.tx_id)
```

Comment:
Action is 'nack' in some repos and 'nack for retry' in others, if they're not functionally different a consistent message would be preferable (though could be raised as a tech debt issue)